pygmt/helpers/testing.py

```diff
@@ -0,0 +1,37 @@
+import os
+import sys
+from pathlib import Path
+
+from matplotlib.testing.compare import compare_images
+
+from ..exceptions import GMTImageComparisonFailure, GMTInvalidInput
+
+
+def check_figures_equal(fig_ref, fig_test, fig_prefix=None, tol=0.0):
```

Comment:
Maybe we should put this into `decorators.py` instead? It's probably easier to do something like this:

```python
@check_figures_equal
def test_some_pygmt_function():
    pass
```

See also https://realpython.com/primer-on-python-decorators/#simple-decorators

Reply:
Feel free to do any changes you want.

Reply:
Ok, I've converted `check_figures_equal` into a decorator at d2ad3f56ba7c2371e4b42c1de998795e4ad7d010 and put the tests as a doctest. It's a bit tricky to make a Python decorator with parameters (see https://stackoverflow.com/questions/5929107/decorators-with-parameters), so still needs a bit more work.